runtime/util/superclass.c

```diff
@@ -29,11 +29,20 @@
 UDATA 
 isSameOrSuperClassOf(J9Class *superClass, J9Class *baseClass)
 {
-	UDATA superClassDepth = J9RAMCLASS_DEPTH(superClass);
 	
-	return ((baseClass == superClass)
-					|| ((J9RAMCLASS_DEPTH(baseClass) > superClassDepth)
-					   && ((J9RAMCLASS_SUPERCLASSES(baseClass)[superClassDepth]) == superClass)));
+	UDATA ret = TRUE;
```

Comment:
Perhaps we need to add a `VMHelper::isSameClass(clazz1, clazz2)` that does the companion test so simplyfy the code